src/Chapter4.hs

```diff
@@ -629,7 +660,13 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM m1 m2 = m1 >>= (\b1 -> (m2 >>= (\b2 -> pure (b1 && b2))))
+{- Fails for `andM (Just False) Nothing`
+      expected: Just False
+      but got: Nothing
+
+      Why is `Just False` expected but for `andM (Just True) Nothing` the result should be `Nothing`?
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 

In your solution, you are only actually using `Applicative` properties (you don't stop as you would in `Monad`).
However, with `Applicative` you are not able to implement a lazier version of `andM`, because of the way `Applicative` is.
Hope it makes sense 🙂

Reply:
Makes sense, thanks for the explanation.